src/record/export/usecases/interface.ts

```diff
@@ -0,0 +1,8 @@
+import type { Writable } from "stream";
+
+export type LocalRecordRepository = {
+  readonly format: string;
+  readonly writer: () => Writer;
+};
+
+type Writer = Writable;
```

Comment:
@ueokande 
We discussed this at the meeting.

- Inheriting Pumpify is not good because it is not expected usage of Pumpify.
- We need to decide what interface we will provide as Stringifier and Writer

We have three options.

1. Node Stream: Weak type (unclear writable/readable type), We can use utils for stream
    - We don't want to extend Pumpify, so we will provide csvStringifierFactory that returns Pumpify instance.
2. Web Stream: Better type. Each stream IF is generics and accepts writable/readable type. We can use utils for stream. Still unstable.
    - Do not concern at this time
3. Custom type: Strong type. We cannot use utils for stream and need to provide stream-compatible IF.
    - The type doesn't inherit or extend Stream
    - We will use a delegation pattern, storing the underlying stream inside the class

As the conclusion:
We choose option 3 (creating custom type).
The type will provide some methods that are compatible with Node Stream API.
- pipe
- write
- AsyncIterator
- end